sympy/functions/special/delta_functions.py

```diff
@@ -31,6 +31,20 @@ class DiracDelta(Function):
 
     5) ``DiracDelta(x,k) = 0``, for all ``x != 0``
 
+    Examples
```

Comment:
I have also been thinking what would be the proper explanation for SymPy. Actually, Dirac delta is a measure, not an ordinary function. A (real) measure is a mapping from a family of sets, typically subsets of reals, to reals. Its value for a set can be considered as the 'integral of the measure over the set'. So, for example, the value of Dirac delta for an interval  is 1 if the interval contains the origin, and 0 otherwise. Only these 'integrated values' are well defined, not the (pointwise) values of the measure itself.

Measures are a subclass of distributions which are functions in a space of so-called test functions defined on the real line. Dirac delta, for example, is simply the function taking a test function  f  to its value  f(0). Distributions are needed to define derivatives of measures. They are not measures in general. The derivative of Dirac delta is the distribution taking a test function  f  to  -f'(0). (This is to formally satisfy the partial integration property: integral of `delta'(x)*f(x)` should be the same as the integral of `-delta(x)*f'(x)`, which is `-f'(0)`